neqo-client/src/main.rs

```diff
@@ -1077,23 +1078,19 @@ fn main() -> Res<()> {
 
         let hostname = format!("{host}");
         let mut token: Option<ResumptionToken> = None;
-        let mut remaining = &urls[..];
         let mut first = true;
         loop {
-            let to_request;
-            if (args.resume && first) || args.download_in_series {
-                to_request = &remaining[..1];
-                remaining = &remaining[1..];
-                if args.resume && first && remaining.is_empty() {
+            let to_request = if (args.resume && first) || args.download_in_series {
+                if args.resume && first && urls.len() < 2 {
```

Comment:
Another annoying thing.  This check could be made outside of the loop, in which case you wouldn't need `first`.

Reply:
Cool. Yep. Simplifies things.

25dd1a83